source/faq.txt

```diff
@@ -148,6 +146,33 @@ When ``MongoClient.close()`` is called by any thread, the driver
 closes all idle sockets and closes all sockets that are in
 use as they are returned to the pool.
 
+
+How do I prevent the "IllegalArgumentException: Invalid BSON field name" error?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Your application may throw this exception when it encounters any of the
+following issues:
+
+- Characters in field names that your MongoDB deployment cannot process
+- Syntax issues in a command
+
+In MongoDB v4.4 and earlier, names that contain characters such as
+periods (``.``) or are prefixed by dollar signs (``$``) are invalid. In
+MongoDB v5.0 and later, you can use field names that contain periods or are
+prefixed by dollar signs, but including them restricts the functionality of the
+documents.
+
+To learn more about how these special characters limit the
+functionality, see the :manual:`General Restrictions </core/dot-dollar-considerations/#general-restrictions>`
+Server documentation.
+
+If you pass an incorrectly formatted document to an operation, your application
```

Comment:
Have you verified this?  I don't think it's true.

Reply:
I verified this one on MongoDB v4.2, Java Sync v4.3 with an incorrectly formatted updateOne call:

```java
collection.updateOne(new Document().append("name", "fizz"), new Document().append("name", "buzz"));
```

This throws the following exception (since it doesn't provide the update operator):
```
Exception in thread "main" java.lang.IllegalArgumentException: Invalid BSON field name name
```

Reply:
See https://github.com/mongodb/mongo-java-driver/commit/6693493aab3ec178feb2470aac1190b52d39e7e7.  The error message changed in 4.8 to make it clearer.